tests/test_ipa_azure.py

```diff
@@ -41,7 +41,7 @@ def setup_method(self, method):
             'provider_config': 'tests/azure/azure.config',
             'running_instance_id': 'fakeinstance',
             'ssh_private_key': 'tests/data/ida_test',
-            'test_dirs': ['tests/data/tests'],
+            'test_dirs': 'tests/data/tests',
```

Comment:
I like the list, then you can log the list of searched directories and it is easier to read rather than a string of comma separated entries. Also I guess we need a "--no-default-dirs" to remove the default directory for tests from the search path.

Reply:
Problem is there's no way to have two n=-1 args/options in command line. That is taken by the tests arg.

E.g.
`ipa test ... test1 test2 etc`

And the config in ini also does not parse lists.

Config:
```
[ipa]
test_dirs = dir1,dir2,dir3
```

Cmd line arg:
`--test-dirs=dir1,dir2,dir3`

So I think the format should be the same between config and cmd line. The most user friendly IMHO is a comma separated list. It could be a python list and parsed by json module. But that doesn't seem very user friendly.

```
[ipa]
test_dirs = ['dir1','dir2','dir3']

json.loads(config.get('ipa', 'test_dirs'))
```

For command line we could use the multiple flag and it would be passed in as a list if we're okay with having that mismatch with format of ini.

```
ipa list --test-dir=dir1 --test-dir=dir2
```

Reply:
That could also be confusing because the option is test-dirs to match ini or test-dir to match command line.

Reply:
At the moment ipa will drop default test directories if a value is provided via config or command line args.

Reply:
Don't confuse internal representation with external representation. Just because we choose handle something as a list internally doe not imply the user has to deal with the list. I agree that config and command line should match and for the interface a comma separated string is the best approach. But as soon as we have the data, str.split(',') turn it into a list

There has to be a way to combine test dirs. I think the intuitive way is that --test-dirs is additive, but we also need a way to drop the default test dirs thus --no-default-test-dirs in the additive behavior case. If we choose to make --test-dirs to have a, as currently implemented, subtractive default behavior, i.e. when used the default dirs are dropped then we'd need a --use-default-test-dirs. Either way and additional option is needed. IMHO additive is more common behavior.

Reply:
Okay that sounds like a better approach.

- By default append to default dirs
- Add --no-default-test-dirs option to drop defaults

The test_dirs is handled as a list internally. It's concatenated, converted and cleaned up when parsing the args/config.